File: abc/131/e.py
from itertools import islice

# 入力
N, K = map(int, input().split())

# スターグラフに適当に辺を追加する
S = (N - 1) * (N - 2) // 2
D = S - K

ans = (
    -1 if K > S else
    '{}\n{}'.format(
        (N - 1) + D,
        '\n'.join(
            '{} {}'.format(u, v)
            for u, v in [
                (1, i)
                for i in range(2, N + 1)
            ] + [
                (u, v)
                for u, v in islice(
                    (
                        (x, y)
                        for x in range(2, N + 1)
                        for y in range(x + 1, N + 1)
                    ),
                    D
                )
            ]
        )
    )
)

# 出力
print(ans)

File: abc/131/e.rs
#![allow(non_snake_case)]
#![allow(unused_variables)]
#![allow(dead_code)]

fn main() {
    let (N, K): (usize, usize) = {
        let mut line: String = String::new();
        std::io::stdin().read_line(&mut line).unwrap();
        let mut iter = line.split_whitespace();
        (
            iter.next().unwrap().parse().unwrap(),
            iter.next().unwrap().parse().unwrap()
        )
    };

    let mut E = (1..N).map(|i| (i, N)).collect::<Vec<_>>();
    let mut k = (N - 2) * (N - 1) / 2;
    for i in 1..N {
        for j in (i + 1)..N {
            if k > K {
                E.push((i, j));
                k -= 1;
            }
        }
    }
    let ans = if k != K {
        "-1".to_string()
    } else {
        format!(
            "{}\n{}",
            E.len(),
            E.iter()
                .map(|&(i, j)| format!("{} {}", i, j))
                .collect::<Vec<_>>()
                .join("\n")
        )
    };

    println!("{}", ans);
}